janusgraph-core/src/main/java/org/janusgraph/graphdb/management/ConfigurationManagementGraph.java

```diff
@@ -335,6 +351,15 @@ private void updateVertexWithProperties(String propertyKey, Object propertyValue
         }
     }
 
+    private void removeVertexProperties(String propertyKey, Object propertyValue, Set<String> set) {
+        final GraphTraversalSource g = getTraversal();
+        if (g.V().has(propertyKey, propertyValue).hasNext()) {
+            final Vertex v = g.V().has(propertyKey, propertyValue).next();
+            set.forEach(key -> v.property(key, null));
+            graph.tx().commit();
+        }
+    }
+
```

Comment:
_(nitpick)_ 
if JanusGraph starts to support `null` values for properties (as TinkerPop does) this logic might potentially break.
I think we could potentially use something like this:
```
private void removeVertexProperties(String propertyKey, Object propertyValue, Set<String> set) {
    final GraphTraversalSource g = getTraversal();
    GraphTraversal<Vertex, Vertex> traversal = g.V().has(propertyKey, propertyValue);
    if (traversal.hasNext()) {
        final Vertex v = traversal.next();
        set.forEach(key -> v.property(key).remove());
        g.tx().commit();
    }
}
```
Or maybe even something simpler (if committing after all properties are removed is OK):
```
private void removeVertexProperties(String propertyKey, Object propertyValue, Set<String> set) {
    final GraphTraversalSource g = getTraversal();
    g.V().has(propertyKey, propertyValue).properties(set.toArray(new String[0])).drop().iterate();
    g.tx().commit();
}
``` 

I didn't check neither of the two methods I suggested in this comment, so can't guarantee they will work, but they should I think.

Reply:
Thanks! I didn't know `null` support was on the radar. I've applied your suggestion (which is cleaner than what I had).

Reply:
It's not on the radar for JanusGraph yet, but TinkerPop already supports `null` values from TinkerPop 3.5.0. Not sure if we are going to support nulls anytime soon, but that's an option for future releases I think.

Related:
https://github.com/JanusGraph/janusgraph/issues/2624 
https://github.com/JanusGraph/janusgraph/pull/2619